sycl/doc/extensions/experimental/sycl_ext_oneapi_tangle.asciidoc

```diff
@@ -0,0 +1,377 @@
+public:
+  using id_type = id<1>;
+  using range_type = range<1>;
+  using linear_id_type = uint32_t;
+  static constexpr int dimensions = 1;
+  static constexpr sycl::memory_scope fence_scope = ParentGroup::fence_scope;
+
+  id_type get_group_id() const;
+
+  id_type get_local_id() const;
+
+  range_type get_group_range() const;
+
+  range_type get_local_range() const;
+
+  linear_id_type get_group_linear_id() const;
+
+  linear_id_type get_local_linear_id() const;
+
+  linear_id_type get_group_linear_range() const;
+
+  linear_id_type get_local_linear_range() const;
+
+  bool leader() const;
+};
+
+}
+----
+
+[source,c++]
+----
+id_type get_group_id() const;
+----
+_Returns_: An `id` representing the index of the tangle.
+
+NOTE: This will always be an `id` with all values set to 0, since there can
+only be one tangle.
+
+[source,c++]
+----
+id_type get_local_id() const;
+----
+_Returns_: An `id` representing the calling work-item's position within
+the tangle.
+
+[source,c++]
+----
+range_type get_group_range() const;
+----
+_Returns_: A `range` representing the number of tangles.
+
+NOTE: This will always return a `range` of 1 as there can only be one
+tangle.
+
+[source,c++]
+----
+range_type get_local_range() const;
+----
+_Returns_: A `range` representing the number of work-items in the tangle.
+
+[source,c++]
+----
+id_type get_group_linear_id() const;
+----
+_Returns_: A linearized version of the `id` returned by `get_group_id()`.
+
+[source,c++]
+----
+id_type get_local_linear_id() const;
+----
+_Returns_: A linearized version of the `id` returned by `get_local_linear_id()`.
+
+[source,c++]
+----
+range_type get_group_linear_range() const;
+----
+_Returns_: A linearized version of the `id` returned by `get_group_range()`.
+
+[source,c++]
+----
+range_type get_local_linear_range() const;
+----
+_Returns_: A linearized version of the `id` returned by `get_local_range()`.
+
+[source,c++]
+----
+bool leader() const;
+----
+_Returns_: `true` for exactly one work-item in the tangle, if the calling
+work-item is the leader of the tangle, and `false` for all other
+work-items in the tangle. The leader of the tangle is guaranteed to
+be the work-item for which `get_local_id()` returns 0.
+
+
+==== Usage examples
+
+A `tangle` can be used in conjunction with constructs like loops and
+branches to safely communicate between all work-items executing the same
+control flow.
+
+NOTE: This differs from the `fragment` returned by `get_opportunistic_group()`
+because a `tangle_group` requires the implementation to track group membership.
+Which group type to use will depend on a combination of
+implementation/backend/device and programmer preference.
+
+[source, c++]
+----
+auto sg = it.get_sub_group();
+
+auto will_branch = sg.get_local_linear_id() % 2 == 0;
+if (will_branch)
+{
+  // wait for all work-items that took the branch to hit the barrier
+  auto inner = sycl::ext::oneapi::experimental::entangle(sg);
+  sycl::group_barrier(inner);
```

Comment:
Why do we need a separate barrier if `entangle` performs a synchronization anyway?

Reply:
You're right, we don't need it any more.  Removed in 9af5173659.